### DISCUSSION TITLE
Catch-all route fails on Vercel (Next.js v14)

### DISCUSSION THREAD

**Discussion by wojtodzio** [vercel/community#4915] [Help]
### Summary

We're using the latest Next.js (14.0.3) with the app router.
We have two catch-all routes on the same level. One is using `generateStaticParams` to generate a few static pages, and the other works as a fallback:
- Route 1: `src/app/[citySlug]/page.tsx` - Uses `generateStaticParams` for static pages like `/new-york-city`.
- Route 2: `src/app/[...railsPathSegments]/page.tsx` - Serves as a fallback route.

Locally, everything operates correctly: `next build` correctly generates all static params for `[citySlug]`, and `next start` routes other paths to `[...railsPathSegments]`.
On Vercel, however, requests that should be routed to `[...railsPathSegments]` result in an error: `Error: Cannot find module '/var/task/web-app/.next/server/app/[citySlug]/page.js'`. Based on logs, the request is correctly routed to `[...railsPathSegments]`, but then it seems to incorrectly try to resolve `[citySlug]/page.js`.

Logs - function seems to be correct, but then, for some reason, it fails while trying to find the wrong module:
![image](https://github.com/vercel/community/assets/4521397/ffdd5ca3-b18a-43b6-bb9e-fa5f5a32ba07)

### Steps to Reproduce

1. Using Next.js version 14.0.3, create two routes: `src/app/[citySlug]/page.tsx` and `src/app/[...railsPathSegments]/page.tsx`.
2. Implement `generateStaticParams` in `src/app/[citySlug]/page.tsx` to generate a page for `/new-york-city`.
3. Locally, run `next build` followed by `next start`. Confirm that:
   - Visiting `/new-york-city` renders `src/app/[citySlug]/page.tsx`.
   - Visiting `/about` renders `src/app/[...railsPathSegments]/page.tsx`.
   Note: Both routes function as expected in the local environment.
4. Deploy the application to Vercel. Post-deployment, verify the behavior of the routes:
   - Visiting `/new-york-city` correctly renders `src/app/[citySlug]/page.tsx`.
   - Visiting `/about` should render `src/app/[...railsPathSegments]/page.tsx`, but it fails with the error: `Error: Cannot find module '/var/task/web-app/.next/server/app/[citySlug]/page.js'`.

**Comment by amyegan** [comment]
Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that I can use to help with debugging?

**Comment by wojtodzio** [comment]
@amyegan Sure: https://github.com/wojtodzio/catch-all-bug-example (relevant commit: https://github.com/wojtodzio/catch-all-bug-example/commit/964362cef1f5f0c7edfa62e2f53103047f5684e0)
Also deployed to Vercel	at https://catch-all-bug-example.vercel.app/
If you build and run it locally, you'll notice that:
- Both `/new-york-city` on localhost and https://catch-all-bug-example.vercel.app/new-york-city render the same thing
- `/about` renders a component correctly on the localhost while https://catch-all-bug-example.vercel.app/about results in a 500 error

**Comment by wojtodzio** [comment]
Hey @amyegan, have you, by any chance, had time to look at the minimal reproducible example I posted above? Is there anything else I could do to help with debugging? Thanks for all the help!

**Reply by amyegan** [reply]
Thanks for sharing that repo! I was able to see what you're talking about and reproduced the same behavior in another app. I just reached out to the Next.js team for their advice, but I don't have any news on that yet.

As a workaround, I was able to get it working by moving the `[citySlug]` route into a sub-directory. In my case, that made the new path `example.com/city/new-york-city`. This removed the dynamic route conflict so the catch-all route started working.
